core/src/test/java/nl/nn/adapterframework/http/rest/ApiListenerServletTest.java

```diff
@@ -1100,10 +1241,29 @@ public void testEndpointDoesNotAcceptHeader(String acceptHeaderValue) throws Exc
 		Response result = service(request);
 
 		// Assert
+		assertFalse(handlerInvoked, "Request Handler should not have been invoked, pre-conditions should have failed and stopped request-processing");
 		assertEquals(406, result.getStatus());
 		assertEquals("endpoint cannot provide the supplied MimeType", result.getErrorMessage());
 	}
 
+	@Test
+	public void testRequestExceptionHandling() throws Exception {
+		// Arrange
+		String uri = "/testThrowsError";
+		new ApiListenerBuilder(uri, Methods.GET, null, null)
+			.withShouldThrow(true)
+			.build();
+
+		HttpServletRequest request = createRequest(uri, Methods.GET, null, null);
+
+		// Act
+		Response result = service(request);
+
+		// Assert
+		assertEquals(500, result.getStatus());
+		assertNotNull(result.getErrorMessage());
```

Comment:
Het is misschien netjes hier ook te kijken of de content-type header gezet is. (er is namelijk een errormessage/data)

Reply:
Wat voor content-type zou gezet moeten zijn?
Waar er een error-message is, zetten en controleren we zelf nergens de content-type header en voor zover ik het weet, is afhankelijk van wat voor error-page de server terug geeft? (Nadat `sendError()` is aangeroepen).